go/vt/vtgate/executor.go

```diff
@@ -198,50 +202,44 @@ func (e *Executor) execute(ctx context.Context, safeSession *SafeSession, sql st
 		}
 		return qr, nil
 	case sqlparser.StmtDDL:
-		return e.handleDDL(ctx, safeSession, sql, bindVars, target, logStats)
+		return e.handleDDL(ctx, safeSession, sql, bindVars, dest, destKeyspace, destTabletType, logStats)
 	case sqlparser.StmtBegin:
-		return e.handleBegin(ctx, safeSession, sql, bindVars, target, logStats)
+		return e.handleBegin(ctx, safeSession, sql, bindVars, destTabletType, logStats)
 	case sqlparser.StmtCommit:
-		return e.handleCommit(ctx, safeSession, sql, bindVars, target, logStats)
+		return e.handleCommit(ctx, safeSession, sql, bindVars, logStats)
 	case sqlparser.StmtRollback:
-		return e.handleRollback(ctx, safeSession, sql, bindVars, target, logStats)
+		return e.handleRollback(ctx, safeSession, sql, bindVars, logStats)
 	case sqlparser.StmtSet:
 		return e.handleSet(ctx, safeSession, sql, bindVars, logStats)
 	case sqlparser.StmtShow:
-		return e.handleShow(ctx, safeSession, sql, bindVars, target, logStats)
+		return e.handleShow(ctx, safeSession, sql, bindVars, dest, destKeyspace, destTabletType, logStats)
 	case sqlparser.StmtUse:
 		return e.handleUse(ctx, safeSession, sql, bindVars)
 	case sqlparser.StmtOther:
-		return e.handleOther(ctx, safeSession, sql, bindVars, target, logStats)
+		return e.handleOther(ctx, safeSession, sql, bindVars, dest, destKeyspace, destTabletType, logStats)
 	case sqlparser.StmtComment:
-		return e.handleComment(ctx, safeSession, sql, bindVars, target, logStats)
+		return e.handleComment(sql)
 	}
 	return nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "unrecognized statement: %s", sql)
 }
 
-func (e *Executor) handleExec(ctx context.Context, safeSession *SafeSession, sql string, bindVars map[string]*querypb.BindVariable, target querypb.Target, logStats *LogStats) (*sqltypes.Result, error) {
-	keyRange, err := parseRange(safeSession.TargetString)
-	if err != nil {
-		return nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "could not parse target %s (%s)", safeSession.TargetString, err.Error())
-	}
-
-	if keyRange != nil || target.Shard != "" {
+func (e *Executor) handleExec(ctx context.Context, safeSession *SafeSession, sql string, bindVars map[string]*querypb.BindVariable, dest key.Destination, destKeyspace string, destTabletType topodatapb.TabletType, logStats *LogStats) (*sqltypes.Result, error) {
```

Comment:
Th usual ordering of the parameters is keyspace, tablet type, and then destination.